src/components/diagrams/singleLineDiagram/position-diagram.js

```diff
@@ -0,0 +1,248 @@
+}));
+
+const PositionDiagram = forwardRef((props, ref) => {
+    const [svg, setSvg] = useState(NoSvg);
+    const svgUrl = useRef('');
+    const svgDraw = useRef();
+    const { snackError } = useSnackMessage();
+    const intlRef = useIntlRef();
+    const svgRef = useRef();
+    const { svgType, disabled } = props;
+
+    const network = useSelector((state) => state.network);
+
+    const currentNode = useSelector((state) => state.currentTreeNode);
+
+    const [loadingState, updateLoadingState] = useState(false);
+
+    const theme = useTheme();
+
+    // using many useState() calls with literal values only to
+    // easily avoid recomputing stuff when updating with the same values
+    const [serverHeight, setServerHeight] = useState(0);
+    const [serverWidth, setServerWidth] = useState(0);
+
+    useEffect(() => {
+        if (props.svgUrl) {
+            updateLoadingState(true);
+            fetchSvg(props.svgUrl)
+                .then((data) => {
+                    setSvg({
+                        svg: data.svg,
+                        metadata: data.metadata,
+                        error: null,
+                        svgUrl: props.svgUrl,
+                    });
+                    updateLoadingState(false);
+                })
+                .catch((errorMessage) => {
+                    console.error(errorMessage);
+                    setSvg({
+                        svg: null,
+                        metadata: null,
+                        error: errorMessage,
+                        svgUrl: props.svgUrl,
+                    });
+                    snackError({
+                        messageTxt: errorMessage,
+                    });
+                    updateLoadingState(false);
+                });
+        } else {
+            setSvg(NoSvg);
+        }
+    }, [props.svgUrl, snackError, intlRef]);
+
+    useLayoutEffect(() => {
+        if (disabled) return;
+
+        if (svg.svg) {
+            let viewboxMaxWidth = MAX_WIDTH_VOLTAGE_LEVEL;
+            let viewboxMaxHeight = MAX_HEIGHT_VOLTAGE_LEVEL;
+            let selectionBackColor = theme.palette.background.paper;
+
+            const sldViewer = new SingleLineDiagramViewer(
+                svgRef.current, //container
+                svg.svg, //svgContent
+                svg.metadata, //svg metadata
+                svgType,
+                0,
+                0,
+                viewboxMaxWidth,
+                viewboxMaxHeight,
+                selectionBackColor //arrows color
+            );
+
+            setServerHeight(sldViewer.getHeight());
+            setServerWidth(sldViewer.getWidth());
+
+            if (svgDraw.current && svgUrl.current === svg.svgUrl) {
+                sldViewer.setViewBox(svgDraw.current.getViewBox());
+            }
+            svgUrl.current = svg.svgUrl;
+            svgDraw.current = sldViewer;
+        }
+    }, [
+        network,
+        svg,
+        currentNode,
```

Comment:
I think we should not use network, svg or currentNode here since they are objects, but it's already done in single-line-diagram.js, to do later then ?

Reply:
I can see it's done several other times, I won't comment them